specs/ChainSpec/Consensus.md

```diff
@@ -0,0 +1,220 @@
+# Consensus
+
+The fields in the Block header relevant to the consensus process are:
+
+```rust
+struct BlockHeader {
+    ...
+    prev_hash: BlockHash,
+    height: BlockHeight,
+    epoch_id: EpochId,
+    last_final_block_hash: BlockHash,
+    approvals: Vec<Option<Signature>>
+    ...
+}
+```
+
+Each block belongs to a particular epoch, and has a particular height. The largest height block accepted by a particular node is its head.
+
+Each epoch has a set of block producers who are assigned to produce blocks in the epoch, and has an assigned block producer for each height in the epoch (to whom we refer as *block proposer at `h`*). The block producers and the assignments in a particular epoch are known from the first block of the *preceding* epoch.
+
+Consecutive blocks do not necessarily have sequential heights. A block at height `h` can have as its previous block a block with height `h-1` or lower.
+
+If a block `B` at some height `h` builds on top of a block `B_prev` that has height `h-1`, and `B_prev` in turn builds on top of a block `B_prev_prev` that has height `h-2`, and all three blocks are in the same epoch, the block `B_prev_prev` is final, and cannot be reverted unless block producers with more than `2/3` cumulative stake deviate from the protocol. 
```

Comment:
I think that with the current epoch switching rule the requirement that all three blocks are in the same epoch is not necessary. Also, replace "more than 2/3" with "more than 1/3".